arch/x86/boot/compressed/sl_main.c

```diff
@@ -278,30 +284,33 @@ void sl_main(u8 *bootparams)
 	u32 data_count, os_mle_len;
 
 	/*
-	 * Currently only Intel TXT is supported for Secure Launch. Testing
-	 * this value also indicates that the kernel was booted successfully
-	 * through the Secure Launch entry point and is in SMX mode.
+	 * Testing this value indicates that the kernel was booted successfully
+	 * through the Secure Launch entry point and is in proper mode.
 	 */
-	if (!(sl_cpu_type & SL_CPU_INTEL))
+	if (!(sl_cpu_type & (SL_CPU_INTEL | SL_CPU_AMD)))
 		return;
 
 	/*
 	 * If enable_tpm fails there is no point going on. The entire secure
 	 * environment depends on this and the other TPM operations succeeding.
 	 */
 	tpm = enable_tpm();
-	if (!tpm)
-		sl_txt_reset(SL_ERROR_TPM_INIT);
+	if (!tpm) {
+		if (sl_cpu_type == SL_CPU_INTEL)
+			sl_txt_reset(TXT_SLERROR_TPM_INIT);
```

Comment:
Please use the value you replaced, SL_ERROR_TPM_INIT, so that you don't have to undo this in your third commit.

Reply:
Done